src/materialized/dependencies.rs

```diff
@@ -14,3 +14,1514 @@
+                    "+--------------+-------------------------------------------------------------------------------------------------------+",
+                ]
+                ,
+                expected_output: vec![
+                    "+---------------------+------+",
+                    "| timestamp           | feed |",
+                    "+---------------------+------+",
+                    "| 2023-01-01T00:00:00 | A    |",
+                    "| 2023-01-02T00:00:00 | B    |",
+                    "| 2023-01-03T00:00:00 | C    |",
+                    "| 2024-12-04T00:00:00 | X    |",
+                    "| 2024-12-05T00:00:00 | Y    |",
+                    "| 2024-12-06T00:00:00 | Z    |",
+                    "+---------------------+------+",
+                ],
+            },
+            TestCase {
+                name: "empty projection",
+                query_to_analyze: "SELECT column1 AS output FROM t3",
+                projection: &[],
+                expected_plan: vec![
+                    "+--------------+-----------------------------+",
+                    "| plan_type    | plan                        |",
+                    "+--------------+-----------------------------+",
+                    "| logical_plan | TableScan: t3 projection=[] |",
+                    "+--------------+-----------------------------+",
+                ],
+                expected_output: vec![
+                    "++",
+                    "++",
+                    "++",
+                ],
+            },
+            TestCase {
+                name: "simple equijoin on year",
+                query_to_analyze: "SELECT * FROM t2 INNER JOIN t3 USING (year)",
+                projection: &["year"],
+                expected_plan: vec![
+                    "+--------------+-------------------------------------+",
+                    "| plan_type    | plan                                |",
+                    "+--------------+-------------------------------------+",
+                    "| logical_plan | Projection: t2.year                 |",
+                    "|              |   Inner Join: t2.year = t3.year     |",
+                    "|              |     TableScan: t2 projection=[year] |",
+                    "|              |     TableScan: t3 projection=[year] |",
+                    "+--------------+-------------------------------------+",
+                ],
+                expected_output: vec![
+                    "+------+",
+                    "| year |",
+                    "+------+",
+                    "| 2023 |",
+                    "| 2023 |",
+                    "| 2023 |",
+                    "| 2024 |",
+                    "| 2024 |",
+                    "| 2024 |",
+                    "+------+",
+                ],
+            },
+            TestCase {
```

Comment:
this is a pretty cool setup (and an extensive body of tests)

Reply:
Probably not enough either! 😅